src/components/utils/hooks/useMessagePreviews.tsx

```diff
@@ -0,0 +1,158 @@
+  const client = useMessageStore((state) => state.client);
+  const setClient = useMessageStore((state) => state.setClient);
+  const isMessagesEnabled = isFeatureEnabled('messages', currentProfile?.id);
+
+  const [stream, setStream] = useState<Stream<Conversation>>();
+  const conversations = useMessageStore((state) => state.conversations);
+  const setConversations = useMessageStore((state) => state.setConversations);
+  const messageProfiles = useMessageStore((state) => state.messageProfiles);
+  const setMessageProfiles = useMessageStore((state) => state.setMessageProfiles);
+  const previewMessages = useMessageStore((state) => state.previewMessages);
+  const setPreviewMessages = useMessageStore((state) => state.setPreviewMessages);
+  const [messagesLoading, setMessagesLoading] = useState<boolean>();
+
+  const peerAddresses = Array.from(conversations.keys());
+  const request = { ownedBy: peerAddresses, limit: 50 };
+  const {
+    loading: profilesLoading,
+    error: profilesError,
+    fetchMore
+  } = useQuery(ProfilesDocument, {
+    variables: {
+      request: request
+    },
+    skip: !currentProfile?.id || peerAddresses.length === 0,
+    onCompleted: async (data) => {
+      if (!data?.profiles?.items?.length) {
+        return;
+      }
+      const profiles = data.profiles.items as Profile[];
+      const pageInfo = data.profiles.pageInfo;
+      const newMessageProfiles = new Map(messageProfiles);
+      for (const profile of profiles) {
+        const peerAddress = (profile.ownedBy as string).toLowerCase();
+        const previousProfile = newMessageProfiles.get(peerAddress);
+        if (!previousProfile?.isDefault) {
+          newMessageProfiles.set(peerAddress, profile);
+        }
+      }
+      setMessageProfiles(newMessageProfiles);
+
+      // Paginate through all profiles for the existing conversations.
+      if (pageInfo.next) {
+        await fetchMore({
+          variables: { request: { ...request, cursor: pageInfo?.next } }
+        });
+      }
+    }
+  });
+
+  useEffect(() => {
+    const initXmtpClient = async () => {
+      if (signer && !client) {
+        const xmtp = await Client.create(signer);
+        setClient(xmtp);
+      }
+    };
+    if (isMessagesEnabled) {
+      initXmtpClient();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [signer]);
+
+  useEffect(() => {
+    if (!isMessagesEnabled || !client) {
+      return;
+    }
+
+    const fetchMostRecentMessage = async (
+      convo: Conversation
+    ): Promise<{ address: string; message?: Message }> => {
+      const peerAddress = convo.peerAddress.toLowerCase();
+      const newMessages = await convo.messages({
+        limit: 1,
+        direction: SortDirection.SORT_DIRECTION_DESCENDING
+      });
+      if (newMessages.length <= 0) {
+        return { address: peerAddress };
+      }
+      return { address: peerAddress, message: newMessages[0] };
+    };
+
+    const listConversations = async () => {
```

Comment:
Similarly, if we are at a point breaking things down, listingConversations and streamingConversations can also be into two different hooks

Reply:
The streaming/listing here is specific to the message preview list so I'd actually prefer to keep it encapsulated in this hook for now. I don't want it to be misused elsewhere (it only fetches one most recent message) and I enjoy that I can share the `fetchMostRecentMessage` when either the list or stream return to ensure everything is handled the same way.

Feel free to push back if you feel strongly -- otherwise I'd rather leave this as-is for now!

Reply:
Ahh makes sense. We can keep it this way